File: java/src/main/java/world/md2html/plugins/IgnorePlugin.java
package world.md2html.plugins;

import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.ObjectReader;
import com.fasterxml.jackson.databind.node.NullNode;
import org.apache.commons.lang3.StringUtils;
import world.md2html.options.argfile.ArgFileParseException;
import world.md2html.options.model.Document;

import java.io.IOException;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.stream.Collectors;

import static world.md2html.utils.JsonUtils.OBJECT_MAPPER;

public class IgnorePlugin extends AbstractMd2HtmlPlugin implements PageMetadataHandler {

    private List<String> markers;
    private List<PageMetadataHandlerInfo> pageLinksHandlers = new ArrayList<>();

    @Override
    public void acceptData(JsonNode data) throws ArgFileParseException {
        assureAcceptDataOnce();
        validateInputDataAgainstSchemaFromResource(data, "plugins/ignore_schema.json");
        JsonNode markersNode = data.get("markers");
        if (markersNode == null || markersNode instanceof NullNode) {
            this.markers = null;
        } else {
            ObjectReader reader = OBJECT_MAPPER.readerFor(new TypeReference<List<String>>() {});
            ArrayList<String> list;
            try {
                list = reader.readValue(markersNode);
            } catch (IOException e) {
                throw new ArgFileParseException("Error reading plugin '" +
                        this.getClass().getSimpleName() + "' data: " + data);
            }
            this.markers = list;
        }
        if (this.markers == null || this.markers.isEmpty()) {
            this.markers = Collections.singletonList("ignore");
        }

        this.pageLinksHandlers = this.markers.stream()
                .map(m -> new PageMetadataHandlerInfo(this, m, false))
                .collect(Collectors.toList());
    }

    @Override
    public boolean isBlank() {
        return this.pageLinksHandlers.isEmpty();
    }

    @Override
    public List<PageMetadataHandlerInfo> pageMetadataHandlers() {
        return this.pageLinksHandlers;
    }

    @Override
    public String acceptPageMetadata(Document document, String marker, String metadata,
            String metadataSection) throws PageMetadataException {

        int contentStart = metadataSection.indexOf(metadata);
        String prefix = metadataSection.substring(0, contentStart - marker.length());
        String suffix = metadataSection.substring(contentStart + metadata.length());
        return prefix + StringUtils.stripStart(metadata, null) + suffix;
    }
}

File: python/plugins/ignore_plugin.py
from pathlib import Path
from typing import List, Union, Dict

from models.document import Document
from plugins.md2html_plugin import Md2HtmlPlugin

MODULE_DIR = Path(__file__).resolve().parent


class IgnorePlugin(Md2HtmlPlugin):

    def __init__(self):
        super().__init__()
        self.markers: List[str] = []
        self.metadata_handlers = []

    def accept_data(self, data):
        self.assure_accept_data_once()
        self.validate_data_with_file(data, MODULE_DIR.joinpath('ignore_schema.json'))
        markers = data.get("markers")
        self.markers = markers if markers else ["ignore"]
        self.metadata_handlers = [(self, m, False) for m in self.markers]

    def is_blank(self) -> bool:
        return not bool(self.metadata_handlers)

    def page_metadata_handlers(self):
        return self.metadata_handlers

    def accept_page_metadata(self, doc: Document, marker: str, metadata_str: str,
                             metadata_section: str,
                             visited_markers: Union[Dict[str, None]] = None):
        content_start = metadata_section.find(metadata_str)
        prefix = metadata_section[:content_start - len(marker)]
        suffix = metadata_section[content_start + len(metadata_str):]
        return prefix + metadata_str.lstrip() + suffix
